compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -702,10 +713,22 @@ impl InitMask {
     #[inline]
     fn set_bit(&mut self, block: usize, bit: usize, new_state: bool) {
         if new_state {
+            self.ensure_blocks(block);
             self.blocks[block] |= 1 << bit;
-        } else {
-            self.blocks[block] &= !(1 << bit);
+        } else if let Some(block) = self.blocks.get_mut(block) {
+            *block &= !(1 << bit);
+        }
+    }
+
+    fn ensure_blocks(&mut self, block: usize) {
+        if block < self.blocks.len() {
+            return;
         }
+        let additional_blocks = block - self.blocks.len() + 1;
+        self.blocks.extend(
+            // FIXME(oli-obk): optimize this by repeating `new_state as Block`.
```

Comment:
I don't quite understand this FIXME. I see it just got moved around but still, it should be clarified or removed IMO.

Reply:
ah, basically instead of filling with uninit and then setting all of them to initialized, we can immediately fill with init.